core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowStateSerializer.java

```diff
@@ -41,7 +38,18 @@ public void serialize(LongDecimalWithOverflowState state, BlockBuilder out)
             long overflow = state.getOverflow();
             long[] decimal = state.getDecimalArray();
             int offset = state.getDecimalArrayOffset();
-            VARBINARY.writeSlice(out, Slices.wrappedLongArray(overflow, decimal[offset], decimal[offset + 1]));
+            long[] buffer = new long[3];
+            long low = decimal[offset + 1];
+            long high = decimal[offset];
+            buffer[0] = low;
+            buffer[1] = high;
+            buffer[2] = overflow;
+            // if high == 0 and overflow == 0 we only write low (bufferLength = 1)
+            // if high != 0 and overflow == 0 we write both low and high (bufferLength = 2)
+            // if overflow != 0 we write all values (bufferLength = 3)
+            int decimalsCount = 1 + (high == 0 ? 0 : 1);
+            int bufferLength = overflow == 0 ? decimalsCount : 3;
```

Comment:
I think `overflow == 0 ? decimalsCount : 3;` might actually turn into branch

Reply:
it may be a branch, but usually well predicted.Even in the mixed case I don't see large impact
```

Benchmark                                                      (decimalSize)  (function)  (groupCount)  (groupIdsInOrder)  (type)  Mode  Cnt      Score     Error  Units
BenchmarkDecimalAggregation.benchmarkEvaluateIntermediateOnly          SHORT         sum       1000000               true    LONG  avgt   10  36281.881 ± 834.705  us/op
BenchmarkDecimalAggregation.benchmarkEvaluateIntermediateOnly           LONG         sum       1000000               true    LONG  avgt   10  56961.139 ± 372.546  us/op
BenchmarkDecimalAggregation.benchmarkEvaluateIntermediateOnly          MIXED         sum       1000000               true    LONG  avgt   10  52997.475 ± 314.941  us/op
```